### PR TITLE
Fix CI issues in generic core libraries

### PR DESCRIPTION
**Pull request by alzimmermsft**
# Description

Fixes CI issues in generic core libraries.

# All SDK Contribution checklist:
- [x] **The pull request does not introduce [breaking changes]**
- [x] **CHANGELOG is updated for new features, bug fixes or other significant changes.**
- [x] **I have read the [contribution guidelines](https://github.com/Azure/azure-sdk-for-java/blob/main/CONTRIBUTING.md).**

## [General Guidelines and Best Practices](https://github.com/Azure/azure-sdk-for-java/blob/main/CONTRIBUTING.md#developer-guide)
- [x] Title of the pull request is clear and informative.
- [x] There are a small number of commits, each of which have an informative message. This means that previously merged commits do not appear in the history of the PR. For more information on cleaning up the commits in your PR, [see this page](https://github.com/Azure/azure-powershell/blob/master/documentation/development-docs/cleaning-up-commits.md).

### [Testing Guidelines](https://github.com/Azure/azure-sdk-for-java/blob/main/CONTRIBUTING.md#building-and-unit-testing)
- [x] Pull request includes test coverage for the included changes.

### CODE REVIEW

File: eng/code-quality-reports/src/main/java/com/azure/tools/checkstyle/checks/ExternalDependencyExposedCheck.java [hunk 1]
```diff
@@ -198,7 +191,23 @@ private boolean isValidClassDependency(String typeName) {
...
+                return true;
+            } else {
+                return false;
+            }
+        } else if ("java.".regionMatches(0, qualifiedName, 0, 5)) {
+            return true;
+        } else if ("reactor.".regionMatches(0, qualifiedName, 0, 8)) {
+            return true;
+        } else if ("org.reactivestreams.".regionMatches(0, qualifiedName, 0, 20)) {
+            return true;
```
Comment (samvaity) [inline eng/code-quality-reports/src/main/java/com/azure/tools/checkstyle/checks/ExternalDependencyExposedCheck.java:208]: We would need tooling updates once we shift the project to it owns repo and tooling to not have reactor and other azure-core references.

File: sdk/generic-sdk-core/generic-core-http-okhttp/pom.xml [hunk 2]
```diff
@@ -70,7 +72,19 @@
     <dependency>
       <groupId>com.squareup.okhttp3</groupId>
       <artifactId>okhttp</artifactId>
-      <version>4.10.0</version> <!-- {x-version-update;com.squareup.okhttp3:okhttp;external_dependency} -->
+      <version>4.12.0</version> <!-- {x-version-update;com.squareup.okhttp3:okhttp;external_dependency} -->
+    </dependency>
+
+    <dependency>
```
Comment (samvaity) [inline sdk/generic-sdk-core/generic-core-http-okhttp/pom.xml:78]: what are these new dep for?

**Review by samvaity — APPROVED** [review]

File: sdk/generic-sdk-core/generic-core-http-okhttp/pom.xml (hunk 2 above)
Comment (samvaity) [reply sdk/generic-sdk-core/generic-core-http-okhttp/pom.xml:78]: Is this not needed in azure-core-http-okhttp then?

Comment (alzimmermsft) [reply sdk/generic-sdk-core/generic-core-http-okhttp/pom.xml:78]: azure-core-http-okhttp doesn't have the additional implementation generic-core-http-okhttp has, at least not yet.

**Pull request merged**